packages/main/src/requests/stream-reader.ts

```diff
@@ -87,20 +105,22 @@ function readFromReader(
             controller.close();
             return;
           }
-          const chunk = new TextDecoder().decode(value);
-          currentText += chunk;
+          currentText += value;
           const match = currentText.match(responseLineRE);
+          // TODO: This needs to be refactored
           if (match) {
             let parsedResponse: GenerateContentResponse;
             try {
               parsedResponse = JSON.parse(match[1]);
+              controller.enqueue(parsedResponse);
+              currentText = currentText.slice(
+                "data: ".length + match[1].length + "\r\n".length,
```

Comment:
@hsubox76 just in case, please check for off-by-one error

Reply:
Checks out.